lib/chef/provisioning/docker_driver/driver.rb

```diff
@@ -227,6 +227,7 @@ def machine_for(machine_spec, machine_options, base_image_name)
           :command => docker_options[:command],
           :env => docker_options[:env],
           :ports => [].push(docker_options[:ports]).flatten,
+          :volumes => [].push(docker_options[:volumes]).flatten.compact,
```

Comment:
Right, it fails for `:ports => [].push(docker_options[:ports]).flatten,` and I have PR for that too. But invoking `compact` function makes it work, so `volumes` gets `[]` value instead of `[null]`.